tests/test_multiaddr.py

```diff
@@ -218,3 +223,47 @@ def test_get_value():
     assert_value_for_proto(a, P_IP4, "0.0.0.0")
     assert_value_for_proto(a, P_UDP, "12345")
     assert_value_for_proto(a, P_UTP, "")
+
+
+def test_bad_initialization_no_params():
+    with pytest.raises(ValueError):
+        Multiaddr()
+
+
+def test_bad_initialization_too_many_params():
+    with pytest.raises(ValueError):
+        Multiaddr("/ip4/0.0.0.0", string_to_bytes("/ip4/0.0.0.0"))
+
+
+def test_get_value_too_many_fields_protocol(monkeypatch):
+    monkeypatch.setattr("multiaddr.multiaddr.Multiaddr.__str__",
+                        lambda ignore: '/udp/1234/5678')
+    a = Multiaddr("/ip4/127.0.0.1/udp/1234")
+    with pytest.raises(ValueError):
+        a.value_for_protocol(P_UDP)
+
+
+def test_multi_addr_str_corruption(monkeypatch):
```

Comment:
pytest has a lot of magic in it, and I certainly don't know it all, so I had to ask. In the past I've used `nose` and `mock.patch`, which make you be pretty explicit about what you're doing. I'm still getting used to pytest.